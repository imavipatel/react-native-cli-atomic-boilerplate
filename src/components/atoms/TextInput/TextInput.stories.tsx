import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { TextInput } from './TextInput';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

export const actions = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask'),
};
storiesOf('TextInput', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <TextInput placeholder="Primary TextInput" />);
