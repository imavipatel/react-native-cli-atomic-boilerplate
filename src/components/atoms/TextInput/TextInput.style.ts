import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  input: {
    justifyContent: 'center',
    alignItems: 'center',
    fontWeight: '600',
    fontSize: 16,
    minWidth: 200,
    minHeight: 50,
    borderColor: '#08949B',
    borderStyle: 'solid',
    borderRadius: 20,
    borderWidth: 1,
    paddingLeft: 17,
    paddingRight: 17,
  },
});
