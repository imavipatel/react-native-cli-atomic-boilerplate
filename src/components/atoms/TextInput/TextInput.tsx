import React from 'react';
import {
  KeyboardTypeOptions,
  TextInput as ReactNativeTextInput,
} from 'react-native';
import { styles } from './TextInput.style';

interface TextInputProps {
  placeholder?: string;
  keyboardType?: KeyboardTypeOptions;
  secureTextEntry?: boolean;
}

export const TextInput: React.FC<TextInputProps> = props => {
  return (
    <ReactNativeTextInput
      style={styles.input}
      placeholder={props.placeholder}
      keyboardType={props.keyboardType}
      secureTextEntry={props.secureTextEntry}
      testID="input"
    />
  );
};
