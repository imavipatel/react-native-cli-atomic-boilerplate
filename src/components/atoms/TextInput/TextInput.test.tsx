import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { TextInput } from './TextInput';

describe('TextInput', () => {
  const spyOnPress = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(<TextInput placeholder="Primary Input" />);

    expect(tree).toMatchSnapshot();
  });
});
