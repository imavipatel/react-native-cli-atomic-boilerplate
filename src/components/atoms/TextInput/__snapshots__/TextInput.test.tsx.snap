// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<TextInput
  placeholder="Primary Input"
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#08949B",
      "borderRadius": 20,
      "borderStyle": "solid",
      "borderWidth": 1,
      "fontSize": 16,
      "fontWeight": "600",
      "justifyContent": "center",
      "minHeight": 50,
      "minWidth": 200,
      "paddingLeft": 17,
      "paddingRight": 17,
    }
  }
  testID="input"
/>
`;
